src/base/worker_pool.h

```diff
@@ -10,18 +10,27 @@ namespace base {
 
 class WorkerPool {
  public:
-  using NetWorker = Fn<void(const Atomic<bool>&, Data&)>;
-  using SimpleWorker = Fn<void(const Atomic<bool>&)>;
+  using NetWorker = Fn<void(WorkerPool*, Data&)>;
```

Comment:
Made this parameter a reference. Unfortunately, `const` cannot be applied here since worker uses `unique_lock`, which only can get mutex by non-const reference.